bin/smash

```diff
@@ -1,95 +1,99 @@
 #!/usr/bin/env coffee
+require('pretty-error').start
 _            = require 'lodash'
 argv         = require('minimist')(process.argv.slice 2)
 chalk        = require 'chalk'
 broadway     = require 'broadway'
-commander    = require 'commander'
+Commander    = require 'commander'
 Orchestrator = require 'orchestrator'
-
-
-
-# Create Broadway instance
-DevTool = new broadway.App()
-DevTool.load = (name) -> DevTool.use require "../src/modules/#{name}"
-
-DevTool.load 'util'
-log = DevTool.util.logger
-
-# Configure Orchestrator instance
-orchestrator = new Orchestrator()
-DevTool.orchestrator = orchestrator
-
-# Register a Task (Orchestrator)
-DevTool._tasks = []
-DevTool.task = (name, deps, fn) ->
-  log.verbose "Registering #{chalk.yellow 'task'} #{chalk.cyan name}"
-  if typeof deps is 'function'
-    fn = deps
-    deps = []
-  DevTool._tasks.push
-    name: name
-    deps: deps
-    fn: fn
-
-# Register a Command (commander.js)
-DevTool.commands = []
-DevTool.command = (command) ->
-  log.verbose "Registering #{chalk.yellow 'command'} '#{chalk.cyan command.cmd}'"
-  DevTool.commands.push command
-
-# Load components
-DevTool.load mod for mod in [
-  # gather information
-  'assumptions'
+Cation       = require 'cation'
+EventEmitter = require 'eventemitter3'
+pkg          = require '../package'
+Module       = require '../src/core/Module'
+
+log = null
+
+class Smasher extends Module
+  @include EventEmitter
+  constructor: (@platform, @user, @util, @project, @helpers, @recipes) ->
+    log = @util.logger
+    @_tasks = []
+    @commands = []
+
+    # register commands and tasks
+    require("../src/modules/#{a}")(@) for a in [
+      'generate'
+      'compile'
+      'build'
+      'test'
+      'serve'
+      'clean'
+      'apps'
+      'document'
+    ]
+
+    # Configure Orchestrator instance
+    @orchestrator = orchestrator = new Orchestrator()
+    orchestrator.add(task.name, task.deps, task.fn) for task in @_tasks
+
+    # Configure Commander.js instance
+    commander = Commander
```

Comment:
I believe you can `new Commander()` for additional instances, but it's one of those libs that gives you a ready-to-use instantiation when you require it.